src/segments/zvm.go

```diff
@@ -0,0 +1,141 @@
+package segments
+
+import (
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/jandedobbeleer/oh-my-posh/src/runtime"
+)
+
+const (
+	// DefaultZigIcon is the default icon used if none is specified
+	DefaultZigIcon = "ZVM"
+
+	// PropertyZigIcon is the property key for the zig icon
+	PropertyZigIcon properties.Property = "zigicon"
+)
+
+// Zvm represents a Zig Version Manager segment
+type Zvm struct {
+	language
+	Version  string // Public for template access
+	ZigIcon  string // Public for template access
+	colorCmd *colorCommand
+}
+
+type colorCommand struct {
+	env runtime.Environment
+}
+
+// colorState represents the ZVM color configuration state
+type colorState struct {
+	enabled bool
+	valid   bool
+}
+
+func (c *colorCommand) detectColorState() colorState {
```

Comment:
why is this necessary? That switch should not reset the entire command output for following commands, right? If so, this is terrible design on their part.